fixtures/test/test_generate_election.py

```diff
@@ -0,0 +1,166 @@
+            if sum(cvr["votes"][contest_name].values()) != votes_allowed
+        ]
+        assert len(invalid_vote_cvrs) == tally["invalid_votes"]
+
+
+def assert_is_valid_manifest(manifest: Manifest, cvrs: List[Ballot]):
+    # Ballot counts should add up
+    assert sum(num_ballots for _, num_ballots in manifest) == len(cvrs)
+
+
+def assert_is_valid_batch_tallies(
+    batch_tallies: BatchTallies,
+    cvrs: List[Ballot],
+    jurisdiction_tally: JurisdictionTally,
+):
+    # Should have all the batches
+    assert {batch_name for batch_name, _ in batch_tallies} == {
+        cvr["batch"]["name"] for cvr in cvrs
+    }
+    # Tallies should add up to the contest totals
+    for choice_name, votes in jurisdiction_tally["tally"].items():
+        assert sum(tally[choice_name] for _, tally in batch_tallies) == votes
+
+
+# We snapshot the results of generating fixtures for this simple election to
+# ensure that the exact calculations stay correct.
+def test_simple_election(snapshot):
+    simple_election_spec: ElectionSpec = {
+        "name": "Simple Election",
+        "contests": [
+            ContestSpec(
+                name="Contest 1",
+                number_of_winners=1,
+                votes_allowed=1,
+                total_ballots_cast=20,
+                tally={"Candidate 1": 15, "Candidate 2": 3, "Candidate 3": 0},
+                jurisdictions=["Jurisdiction 1", "Jurisdiction 2"],
+            ),
+        ],
+        "jurisdictions": [
+            JurisdictionSpec(name="Jurisdiction 1"),
+            JurisdictionSpec(name="Jurisdiction 2"),
+            JurisdictionSpec(name="Jurisdiction 3"),
+        ],
+    }
+
+    jurisdiction_tallies = split_contest_tallies_across_jurisdictions(
+        simple_election_spec
+    )
+    assert_is_valid_jurisdiction_tallies(jurisdiction_tallies, simple_election_spec)
+    snapshot.assert_match(jurisdiction_tallies)
+
+    for tallies in jurisdiction_tallies.values():
+        cvrs = list(generate_cvrs(tallies, simple_election_spec["contests"]))
+        assert_is_valid_cvrs(cvrs, tallies)
+        snapshot.assert_match(cvrs)
+
+        manifest = cvrs_to_manifest(cvrs)
+        assert_is_valid_manifest(manifest, cvrs)
+        snapshot.assert_match(manifest)
+
+        for contest in simple_election_spec["contests"]:
+            if contest["name"] in tallies:
+                batch_tallies = cvrs_to_batch_tallies(cvrs, contest)
+                assert_is_valid_batch_tallies(
+                    batch_tallies, cvrs, tallies[contest["name"]]
+                )
+                snapshot.assert_match(batch_tallies)
+
+
+# We don't snapshot the results of generating fixtures for this election, since
+# we may want to change the spec for other reasons. However, we can still check
+# that the resulting data is valid (using property assertions). It's important
+# to test with a slightly larger election because there are edge cases that only
+# show up at a slightly larger scale.
```

Comment:
Sanity check: This comment mentions testing with a "slightly larger" election, and this test is called "test_small_election" - is this basically a TODO to add a test for a larger election?

Reply:
Haha I see how that is confusing. `small-election.spec.json` is one of the real fixtures. It's larger than `simple_election`, but still on the small side. I'll update this comment to clarify